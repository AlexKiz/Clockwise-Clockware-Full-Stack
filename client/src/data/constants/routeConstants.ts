/* eslint-disable max-len */
/* eslint-disable no-unused-vars */
export enum URL {
    CITY = '/city',
    LOGIN = '/login',
    MASTER = '/master',
    USER = '/user',
    CLOCK = '/clocks',
    ORDER = '/order',
    AVAILABLE_MASTER = '/availableMasters',
    CITY_FOR_ORDER = '/cityForOrder',
    ORDER_FOR_RATE = '/orderForRate',
	ORDER_FOR_UPDATE = '/orderForUpdate',
	CITY_FOR_UPDATE = '/cityForUpdate',
    RATED_ORDER = '/ratedOrder',
    ORDERS_RATING = '/ordersRating',
	REGISTRATION = '/user/registration',
	VERIFY = '/user/verify',
	COMPLETE_ORDER = '/completeOrder',
	EXPORT_XLSX = '/exportXLSX',
	STRIPE = 'stripe',
	TOTAL_ORDERS_CHART = 'totalOrdersChart',
	TOTAL_ORDERS_CITIES_PIE_CHART = '/totalOrdersForCitiesPieChart',
    TOTAL_ORDERS_MASTERS_PIE_CHART = '/totalOrdersForMastersPieChart',
	MASTERS_STATISTICS_TABLE = '/mastersStatisticsTable',
	ORDERS_FOR_CALENDAR = '/ordersForCalendar',
}

export enum RESOURCE {
    ADMIN = 'admin',
	MASTER = 'master',
	CLIENT = 'client',
    CITY_CREATE = 'city-create',
    CITIES_LIST = 'cities-list',
    ORDER_CREATE = 'order-create',
    ORDERS_LIST = 'orders-list',
    MASTER_CREATE = 'master-create',
    MASTERS_LIST = 'masters-list',
    USER_CREATE = 'user-create',
    USERS_LIST = 'users-list',
    LOGIN = 'login',
    RATE = 'rate',
    REGISTRATION = 'registration-form',
	VERIFICATION = 'email-verification',
	STATISTICS = 'statistics',
	CALENDAR = 'calendar',
}

export enum PARAM {
    MASTER_ID = 'masterIdParam',
    MASTER_NAME = 'masterNameParam',
    USER_ID = 'userIdParam',
    USER_NAME = 'userNameParam',
    USER_EMAIL = 'userEmailParam',
    CITY_ID = 'cityIdParam',
    CITY_NAME = 'cityNameParam',
    ORDER_ID = 'orderIdParam',
    ORDER_DATE = 'orderDateParam',
    ORDER_TIME = 'orderTimeParam',
    CLOCK_ID = 'clockIdParam',
    RATING_ID = 'ratingIdentificatorParam',
	HASH_VERIFY = 'hashVerify',
	GENERATED = 'generated'
}

export const ROUTES = {
	ORDER_FORM: {
		path: '/',
		exact: true,
	},
	REGISTRATION: {
		path: `/${RESOURCE.REGISTRATION}`,
	},
	ADMIN_MASTERS_LIST: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.MASTERS_LIST}`,
	},
	ADMIN_MASTER_CREATE: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.MASTER_CREATE}/:${PARAM.MASTER_ID}?/:${PARAM.MASTER_NAME}?`,
	},
	ADMIN_USERS_LIST: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.USERS_LIST}`,
	},
	ADMIN_USER_CREATE: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.USER_CREATE}/:${PARAM.USER_ID}?/:${PARAM.USER_NAME}?/:${PARAM.USER_EMAIL}?`,
	},
	ADMIN_CITIES_LIST: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.CITIES_LIST}`,
	},
	ADMIN_CITY_CREATE: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.CITY_CREATE}/:${PARAM.CITY_NAME}?`,
	},
	ADMIN_ORDERS_LIST: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.ORDERS_LIST}`,
	},
	ADMIN_STATISTICS: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.STATISTICS}`,
	},
	MASTER_ORDERS_LIST: {
		path: `/${RESOURCE.MASTER}/${RESOURCE.ORDERS_LIST}`,
	},
	MASTER_CALENDAR: {
		path: `/${RESOURCE.MASTER}/${RESOURCE.CALENDAR}`,
	},
	CLIENT_ORDERS_LIST: {
		path: `/${RESOURCE.CLIENT}/${RESOURCE.ORDERS_LIST}`,
	},
	ADMIN_ORDER_CREATE: {
		path: `/${RESOURCE.ADMIN}/${RESOURCE.ORDER_CREATE}/:${PARAM.ORDER_ID}?`,
	},
	LOGIN: {
		path: `/${RESOURCE.LOGIN}`,
	},
	RATE_ORDER: {
		path: `/${RESOURCE.RATE}/:${PARAM.RATING_ID}?`,
	},
	USER_VERIFY: {
		path: `/${RESOURCE.VERIFICATION}/:${PARAM.HASH_VERIFY}?/:${PARAM.GENERATED}?`,
	},
};
