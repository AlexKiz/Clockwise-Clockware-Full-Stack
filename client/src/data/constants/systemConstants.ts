/* eslint-disable max-len */
export const OPENING_HOURS: string[] = ['09:00', '10:00', '11:00', '12:00', '13:00', '14:00', '15:00', '16:00', '17:00', '18:00'];
export const ACCESS_TOKEN = 'accessToken';

export const VALID = {
	NAME: new RegExp(/^([(A-Za-zА-Яа-я]{3,49})$|^([A-Za-zА-Яа-я]{3,49}[\s]{1}[A-Za-zА-Яа-я]{3,50})$/),
	EMAIL: new RegExp(/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/),
	CITY: new RegExp(/^[A-Za-zА-Яа-я]{3,100}$|^[A-Za-zА-Яа-я]{3,49}[-\s]{1}[A-Za-zА-Яа-я]{3,50}$/),
	PASSWORD: new RegExp(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&]).{0,}$/),
};

export enum ROLE {
	ADMIN = 'admin',
    MASTER = 'master',
	CLIENT = 'client',
}

export enum SORTED_FIELD {
	ID = 'id',
	NAME = 'name',
	EMAIL = 'email',
	RATING = 'rating',
	CLOCK_SIZE = 'clock.size',
	USER_NAME = 'user.name',
	USER_EMAIL = 'user.email',
	CITY_NAME = 'city.name',
	MASTER_NAME = 'master.name',
	START_WORK_ON = 'startWorkOn',
	END_WORK_ON = 'endWorkOn',
	SMALL_CLOCKS = 'smallClocks',
	MEDIUM_CLOCKS = 'mediumClocks',
	LARGE_CLOCKS = 'largeClocks',
	COMPLETED = 'completed',
	UNCOMPLETED = 'uncompleted',
	EARNED_AMOUNT = 'earnedAmount'
}

export enum SORTING_ORDER {
	ASC = 'asc',
	DESC = 'desc'
}
