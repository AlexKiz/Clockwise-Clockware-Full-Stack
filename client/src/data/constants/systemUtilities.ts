/* eslint-disable require-jsdoc */
import {useCallback, useRef} from 'react';

export function debouncer(callback, delay: number) {
	const timer = useRef<null | NodeJS.Timer>(null);

	const debouncedCallback = useCallback((...args) => {
		if (timer.current) {
			clearTimeout(timer.current);
		}

		timer.current = setTimeout(() => {
			callback(...args);
		}, delay);
	}, [callback, delay]);

	return debouncedCallback;
}
