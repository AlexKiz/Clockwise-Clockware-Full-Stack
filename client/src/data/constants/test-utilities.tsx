import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter, Route} from 'react-router-dom';


export const renderWithRouter = ({component, path, exact = false}) => (
	render(
		<MemoryRouter initialEntries={[path]}>
			<Route path={`${path}`} exact={exact}>
				{component}
			</Route>
		</MemoryRouter>,
	)
);

export const rerenderWrapper = ({component, path, exact = false}) => {
	return (
		<MemoryRouter initialEntries={[path]}>
			<Route path={`${path}`} exact={exact}>
				{component}
			</Route>
		</MemoryRouter>
	);
};

export * from '@testing-library/react';
