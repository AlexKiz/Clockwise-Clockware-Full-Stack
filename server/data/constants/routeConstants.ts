/* eslint-disable no-unused-vars */
export enum URL {
    CITY = 'city',
    LOGIN = 'login',
    MASTER = 'master',
    USER = 'user',
    CLOCKS = 'clocks',
    ORDER = 'order',
    AVAILABLE_MASTER = 'availableMasters',
    CITY_FOR_ORDER = 'cityForOrder',
    ORDER_FOR_RATE = 'orderForRate',
    RATED_ORDER = 'ratedOrder',
    API = 'api'
}

export enum RESOURCE {
    RATE = 'rate'
}
