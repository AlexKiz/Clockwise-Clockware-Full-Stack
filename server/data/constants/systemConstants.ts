/* eslint-disable max-len */
export const VALID = {
	CITY_NAME: new RegExp(/^[A-Za-zА-Яа-я]{3,100}$|^[A-Za-zА-Яа-я]{3,49}[-\s]{1}[A-Za-zА-Яа-я]{3,50}$/),
	MASTER_NAME: new RegExp(/^[A-Za-zА-Яа-я]{3,49}$|^[A-Za-zА-Яа-я]{3,49}[\s]{1}[A-Za-zА-Яа-я]{3,50}$/),
	USER_NAME: new RegExp(/^[A-Za-zА-Яа-я]{3,49}$|^[A-Za-zА-Яа-я]{3,49}[\s]{1}[A-Za-zА-Яа-я]{3,50}$/),
	USER_EMAIL: new RegExp(/^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/),
	DATE: new RegExp(/(\d{4}[-](0[1-9]|1[0-2])[-]([0-2]{1}\d{1}|3[0-1]{1})|([0-2]{1}\d{1}|3[0-1]{1})[-](0[1-9]|1[0-2])[-]\d{4})[T]([0-2]{1}\d{1}[:][0-5]{1}\d{1})[:]([0-5]{1}\d{1})[.]([0-5]{1}\d{2}[Z])/),
};

export const METHOD = 'OPTIONS';
