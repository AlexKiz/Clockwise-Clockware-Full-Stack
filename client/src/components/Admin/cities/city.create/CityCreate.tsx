import axios from 'axios';
import React, {useState, useEffect, FC} from 'react';
import {useParams, useHistory} from 'react-router-dom';
import './city-create-form.css';
import {City, Params} from '../../../../data/types/types';
import {CityCreateProps} from './componentConstants';
import {RESOURCE, URL} from '../../../../data/constants/routeConstants';


const CityCreate: FC<CityCreateProps> = () => {
	const history = useHistory();

	const [cityName, setCityName] = useState<string>('');
	const [cityId, setCityId] = useState<number>(0);

	const {cityIdParam, cityNameParam} = useParams<Params>();


	useEffect(() => {
		if (cityIdParam) {
			setCityId(Number(cityIdParam));
			setCityName(cityNameParam);
		}
	}, []);


	const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
		event.preventDefault();
		if (!cityIdParam) {
			axios.post<City>(URL.CITY,
				{
					id: cityId,
					name: cityName,
				}).then(() => {
				alert('City has been created');
				history.push(`/${RESOURCE.ADMIN}/${RESOURCE.CITIES_LIST}`);
			}).catch((error) => {
				if (Number(error.response.status) === 400) {
					alert(error.response.data[0]);
					setCityName('');
				}
			});
		} else {
			axios.put<City>(URL.CITY, {
				id: cityId,
				name: cityName,
			}).then(() => {
				alert('City has been updated');
				history.push(`/${RESOURCE.ADMIN}/${RESOURCE.CITIES_LIST}`);
			}).catch((error) => {
				alert(error.response.data);
				setCityName(cityNameParam);
			});
		}
	};

	return (

		<div className='container-form'>

			<form className='form' onSubmit = {onSubmit}>

				<div>

					<div className='form-section'>
						<div className='form-input__label'>
							<label>Enter city name:</label>
						</div>
						<input
							type = "text"
							placeholder = "Name"
							pattern='^[A-Za-zА-Яа-я]{3,100}$|^[A-Za-zА-Яа-я]{3,49}[-\s]{1}[A-Za-zА-Яа-я]{3,50}$'
							title='City name must be at least 3 letter and alphabetical only'
							value = {cityName}
							onChange = {(cityNameEvent) =>setCityName(cityNameEvent.target.value)}
						></input>
					</div>

					<div className='form-button'>
						<button type = 'submit'>Submit</button>
					</div>

				</div>

			</form>
		</div>
	);
};

export default CityCreate;
