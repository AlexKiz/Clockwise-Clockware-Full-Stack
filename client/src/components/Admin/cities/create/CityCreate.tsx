/* eslint-disable linebreak-style */
/* eslint-disable max-len */
import axios from 'axios';
import React, {FC, useState} from 'react';
import {useParams, useHistory} from 'react-router-dom';
import classes from './city-create-form.module.css';
import {AlertNotification, City, Params} from '../../../../data/types/types';
import {CityCreateProps, validate} from './componentConstants';
import {RESOURCE, URL} from '../../../../data/constants/routeConstants';
import {useFormik} from 'formik';
import {Button, Stack, TextField} from '@mui/material';
import AlertMessage from 'src/components/Notification/AlertMessage';


const CityCreate: FC<CityCreateProps> = () => {
	const history = useHistory();

	const {cityIdParam, cityNameParam} = useParams<Params>();

	const [alertOptions, setAlertOptions] = useState<AlertNotification>({
		notify: false,
		type: 'success',
		message: '',
	});


	const isOpen = (value:boolean) => {
		setAlertOptions({...alertOptions, notify: value});
	};

	const formik = useFormik({
		initialValues: {
			cityName: cityNameParam || '',
			cityId: Number(cityIdParam || 0),
		},
		validate,
		onSubmit: async (values) => {
			if (!cityIdParam) {
				await axios.post<City>(URL.CITY,
					{
						id: values.cityId,
						name: values.cityName,
					}).then(() => {
					setAlertOptions({message: 'City has been created', type: 'success', notify: true});
					history.push(`/${RESOURCE.ADMIN}/${RESOURCE.CITIES_LIST}`);
				}).catch((error) => {
					if (Number(error.response.status) === 400) {
						setAlertOptions({message: error.response.data, type: 'error', notify: true});
						values.cityName = '';
					}
				});
			} else {
				await axios.put<City>(URL.CITY, {
					id: values.cityId,
					name: values.cityName,
				}).then(() => {
					setAlertOptions({message: 'City has been updated', type: 'success', notify: true});
					history.push(`/${RESOURCE.ADMIN}/${RESOURCE.CITIES_LIST}`);
				}).catch((error) => {
					setAlertOptions({message: error.response.data, type: 'error', notify: true});
					values.cityName = cityNameParam;
				});
			}
		},
	});


	return (
		<div>

			<div className={classes.container_form}>

				<form className={classes.form} onSubmit = {formik.handleSubmit}>

					<Stack direction="column" justifyContent="center" spacing={1.5}>
						<div className={classes.form_section}>
							<div className={classes.form_input__label}>
								<label>Enter city name:</label>
							</div>
							<TextField
								id="cityName"
								name="cityName"
								label="City name"
								placeholder="Name"
								variant="filled"
								size="small"
								margin="dense"
								fullWidth
								value={formik.values.cityName}
								onChange={formik.handleChange}
								onBlur={formik.handleBlur}
								error={formik.touched.cityName && Boolean(formik.errors.cityName)}
								helperText={formik.touched.cityName && formik.errors.cityName}
								required
							/>
						</div>

						<div className={classes.form_section}>
							<Button
								variant="contained"
								type="submit"
								className={classes.form_btn}
								style={ {fontSize: 18, backgroundColor: 'green', borderRadius: 15} }
							>
								Submit
							</Button>
						</div>
					</Stack>

				</form>
				{
					alertOptions.notify && <AlertMessage alertType={alertOptions.type} message={alertOptions.message} isOpen={isOpen} notify={alertOptions.notify}/>
				}
			</div>

		</div>
	);
};

export default CityCreate;
