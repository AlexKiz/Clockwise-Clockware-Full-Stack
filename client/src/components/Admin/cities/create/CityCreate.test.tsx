import React from 'react';
import CityCreate from './CityCreate';
import {fireEvent, act, cleanup} from '@testing-library/react';
import {renderWithRouter} from '../../../../data/constants/test-utilities';


it('matches snapshot', () => {
	const {asFragment} = renderWithRouter({
		component: <CityCreate />,
		path: '/admin/city-create',
	});
	expect(asFragment()).toMatchSnapshot();
});

describe('City Create', () => {
	afterEach(() => {
		cleanup();
	});
	const mockSubmit = jest.fn();
	const cityCreateValues = {
		name: '',
	};
	jest.mock('react-router-dom', () => ({
		useHistory: () => ({
			push: jest.fn(),
		}),
	}));

	it('shows input with empty value', () => {
		const {getByTestId} = renderWithRouter({
			component: <CityCreate />,
			path: '/admin/city-create',
		});
		expect((getByTestId('city-name-input') as HTMLInputElement).value).toBe('');
	});

	it('shows render component with initial value', () => {
		const {getByTestId} = renderWithRouter({
			component: <CityCreate />,
			path: '/admin/city-create/Dnipro',
		});
		expect((getByTestId('city-name-input') as HTMLInputElement).value).toBe('Dnipro');
	});
});
