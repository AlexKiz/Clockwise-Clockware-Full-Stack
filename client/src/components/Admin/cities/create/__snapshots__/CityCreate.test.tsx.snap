// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <header>
      <div
        class="wrapper_header"
      >
        <div
          class="wrapper_logo"
        >
          <a
            href="/"
          >
            <div
              class="inner_logo"
            >
              <div
                class="inner_logo_img"
              >
                <div
                  class="logo_img1a"
                >
                  <div
                    class="logo_img1b"
                  >
                    <div
                      class="logo_img1c"
                    />
                  </div>
                </div>
              </div>
              <div
                class="logo_text1d"
              />
            </div>
          </a>
        </div>
        <nav>
          <ul
            class="nav__links"
          >
            <li
              class=""
            >
              <a
                href="/"
              >
                Make order
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/login"
              >
                Log in
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/registration-form"
              >
                Sign up
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary  css-1s4qybw-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  role="button"
                  tabindex="0"
                >
                  English
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value="en"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <div
      class="container_form"
    >
      <form
        class="form"
      >
        <div
          class="css-adynvz-MuiStack-root"
        >
          <div
            class="form_section"
          >
            <div
              class="form_input__label"
            >
              <label
                class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
              >
                Enter city name:
              </label>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="name"
                id="name-label"
              >
                City name
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-10i04qz-MuiInputBase-root-MuiFilledInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                  data-testid="city-name-input"
                  id="name"
                  name="name"
                  placeholder="Name"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="form_section"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form_btn css-sghohy-MuiButtonBase-root-MuiButton-root"
              style="font-size: 18px; background-color: green; border-radius: 15px;"
              tabindex="0"
              type="submit"
            >
              Submit
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
